google/gax/__init__.py

```diff
@@ -482,3 +496,180 @@ def __next__(self):
+
+           This method will wait up to timeout seconds. If the call hasn't
+           completed in timeout seconds, then a RetryError will be raised.
+           timeout can be an int or float. If timeout is not specified or None,
+           there is no limit to the wait time.
+        """
+        if not self._poll(timeout).HasField('response'):
+            raise GaxError(self._operation.error.message)
+
+        return _from_any(self._result_type, self._operation.response)
+
+    def exception(self, timeout=None):
+        """Similar to result(), except returns the exception if any."""
+        if self._poll(timeout).HasField('error'):
+            return self._operation.error
+
+        return None
+
+    def cancelled(self):
+        """Return True if the call was successfully cancelled."""
+        self._get_operation()
+        return (self._operation.HasField('error') and
+                self._operation.error.code == code_pb2.CANCELLED)
+
+    def done(self):
+        """Issues OperationsClient.get_operation and returns value of
+           Operation.done.
+        """
+        return self._get_operation().done
+
+    def add_done_callback(self, fn):  # pylint: disable=invalid-name
+        """Enters a polling loop on OperationsClient.get_operation, and once the
+           operation is done or cancelled, calls the function with this
+           _OperationFuture. Added callables are called in the order that they
+           were added.
+        """
+        self._done_clbks.append(fn)
+        if self._process is None:
+            self._process = mp.Process(target=self._execute_clbks)
+            self._process.start()
+        elif not self._process.is_alive() and self._operation.done:
+            _try_callback(self, fn)
+
+    def operation_name(self):
+        """Returns the value of Operation.name from the last call to
+           OperationsClient.get_operation (or if only the initial API call has
+           been made, the name from that first call).
+        """
+        return self._operation.name
+
+    def metadata(self):
+        """Returns the value of Operation.metadata from the last call to
+           OperationsClient.get_operation (or if only the initial API call has
+           been made, the metadata from that first call).
+        """
+        if self._operation.metadata is None:
+            return None
+
+        return _from_any(self._metadata_type, self._operation.metadata)
+
+    def last_operation_data(self):
+        """Returns the data from the last call to OperationsClient.get_operation
+           (or if only the initial API call has been made, the data from that
+           first call).
+        """
+        return self._operation
+
+    def _get_operation(self):
+        if not self._operation.done:
+            self._operation = self._client.get_operation(
+                self._operation.name, self._call_options)
+
+        return self._operation
+
+    def _done_check(self, _timeout):
```

Comment:
A couple of things I'm unsure about:

I think this would be better as a closure inside _poll, but I'm curious whether it is more pythonic to use a closure or to have it be a private instance method as you have done.

I would use `_`, instead of `_timeout` for the parameter name since the parameter is unused, but I am also unsure if that convention is pythonic.